scripts/generate_error_code.hs

```diff
@@ -20,166 +20,127 @@ data ErrorItem = ErrorItem
-      , eiCode = 305
-      , eiDesc = "Flip for `fee_growth_outside` failed. (This is an invariant of the contract)."
-      }
-  , ErrorItem
-      { eiLabel = "internal_306"
-      , eiCode = 306
-      , eiDesc = "Thrown when `(p.dx - dx_consummed)` is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_liquidity_below_zero_err"
-      , eiCode = 308
-      , eiDesc = "Liquidity went below zero."
-      }
-  , ErrorItem
-      { eiLabel = "internal_309"
-      , eiCode = 309
-      , eiDesc = "Thrown when `(p.dx - r.dx)` is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_insufficient_balance_err"
-      , eiCode = 310
-      , eiDesc = "Contract does not have enough liquidity to execute the swap."
-      }
-  , ErrorItem
-      { eiLabel = "internal_311"
-      , eiCode = 311
-      , eiDesc = "Thrown when `s.i_c >= key.hi.i` and `(s.fee_growth.x - tick_hi.fee_growth_outside.x)` (or `y`) is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_312"
-      , eiCode = 312
-      , eiDesc = "Thrown when `s.i_c < key.hi.i` and `(s.fee_growth.x - tick_lo.fee_growth_outside.x)` (or `y`) is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_position_underflow_err"
-      , eiCode = 313
-      , eiDesc = "Number of positions underflow."
-      }
-  , ErrorItem
-      { eiLabel = "internal_314"
-      , eiCode = 314
-      , eiDesc = "Thrown when `(s.fee_growth.x - f_a.x - f_b.x)` is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_315"
-      , eiCode = 315
-      , eiDesc = "Thrown when `(s.fee_growth.y - f_a.y - f_b.y)` is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_316"
-      , eiCode = 316
-      , eiDesc = "Thrown when `(fee_growth_inside.x - position.fee_growth_inside_last.x)` is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_317"
-      , eiCode = 317
-      , eiDesc = "Thrown when `(fee_growth_inside.y - position.fee_growth_inside_last.y)` is not nat."
-      }
-  , ErrorItem
-      { eiLabel = "internal_sqrt_price_grow_err_1"
-      , eiCode = 318
-      , eiDesc = "Thrown when `s.i_c < i_l.i` and the `sqrt_price` happened not to grow monotonically with tick indices (This is an invariant of the contract)."
-      }
-  , ErrorItem
-      { eiLabel = "internal_sqrt_price_grow_err_2"
-      , eiCode = 319
-      , eiDesc = "Thrown when `i_l.i <= s.i_c && s.i_c < i_u.i` and the `sqrt_price` happened not to grow monotonically with tick indices (This is an invariant of the contract)."
-      }
-  , ErrorItem
-      { eiLabel = "internal_impossible_err"
-      , eiCode = 320
-      , eiDesc = "Generic impossible error."
-      }
-  , ErrorItem
-      { eiLabel = "internal_negative_seconds_outside_err"
-      , eiCode = 321
-      , eiDesc = "Thrown when `seconds_outside` is negative."
-      }
+internalErrors = errorsEnumerate 300
+  -- New errors must be added to the _end_ of this list.
+  -- To remove an error, replace it with 'removedError'.
```

Comment:
This means we keep maintaining a list of removed errors even we no longer need them right?

Reply:
I see that we replace it with `removedError` to maintain the order of error codes. I guess this is alright.

Reply:
Yes, but only after the first release, so such errors should be rare. My impression is this is better than assigning codes manually.